spec.bs

```diff
@@ -1685,23 +1770,33 @@ To <dfn>convert a modifier to a string</dfn> given a [=part/modifier=] |modifier
   1. Set |result|["{{URLPatternInit/search}}"] to |search|.
   1. Set |result|["{{URLPatternInit/hash}}"] to |hash|.
   1. Let |baseURL| be null.
-  1. If |init|["{{URLPatternInit/baseURL}}"] is not null:
+  1. If |init|["{{URLPatternInit/baseURL}}"] [=map/exists=]:
+    <div class="note">
+      The base URL can be used to supply additional context, but for each component, if |init| includes a component which is at least as specific as one in the base URL, none is inherited.
+
+      A component is more specific if it appears later in one of the following two lists (which are very similar to the order they appear in the URL syntax):
+
+      * protocol, hostname, port, pathname, search, hash
+      * protocol, hostname, port, username, password
+
+      Username and password are also never inherited from a base URL when constructing a {{URLPattern}}. (They are, however, inherited from the base URL supplied as an argument to {{URLPattern/test()}} or {{URLPattern/exec()}}.)
```

Comment:
```suggestion
      Username and password are also never inherited from a base URL when constructing a {{URLPattern}}. (They are, however, inherited from the base URL when parsing a URL from the string supplied as an argument to {{URLPattern/test()}} or {{URLPattern/exec()}}.)
```

Reply:
Added "when parsing a URL". You can actually supply a dictionary rather than a string if you want, and this behavior also applies there. e.g. `urlPattern.test({pathname: "/world"}, "https://user:pass@example.com/")` is a thing you can do.